easierdata/Hot-Cold-Simulation
# Sd configuration enhancement

This PR contains feature enhancements focused on storing and accessing project environment variables.
- Added the [python-dotenv](https://pypi.org/project/python-dotenv/) as a dev dependency.  This library is used to easily import project environment variables.
- Environment variables are no longer hardcoded for things such as user credentials.
- Project variables are maintained in the `defaults.py` file. Default values can be applied to variables as needed.
- New CLI script to generate .env files.

# How to Test:
1. Run the command `poetry install` to install the latest library additions.
2. Run the command `poetry run generate-config --help` to learn more about the command and its usage.
3. Run the command `poetry run generate-config` to generate the configuration files.
4. A directory named `config` should appear at the project root.  Inside the directory is a file named `database.env` which contains variables that a user can populate.
